frontends/web/src/app.tsx

```diff
@@ -116,14 +113,33 @@ class App extends Component<Props, State> {
             }
         });
 
-        getAccounts()
-            .then(accounts => this.setState({ accounts }, this.maybeRoute))
-            .catch(console.error);
+        getDeviceList()
+        .then(devices => {
+            return getAccounts()
+            .then(accounts => {
+                this.setState({ accounts, devices }, this.maybeRoute);
+            });
+        })
+        .catch(console.error);
 
```

Comment:
it is not so nice that we now have to both manually call the endpoints and also subscribe the endpoints, each time doing similar/the same things after it finishes.

the previous subscribe decorator took care of both. can your new subscribe api also take care of both at the same time?

food for thought, not in this PR.

Reply:
Currently:
- App.tsx and devices.jsx (old BB01) are the only places where we subscribe and fetch `devices/registered`.
- Also App.tsx is the only place where it subscribes to syncAccountsList, which are then passed as props to the children.

> each time doing similar/the same things after it finishes

Was doing the same/similar thing, but not anymore with this PR. The routing fix in this PR needs to compare the current devices list with the new one, so it can't be in mayberoute.

Personally I don't like mayberoute it was the cause of various routing bugs in the past and has to maintain a centralized list of exception routing for different views, it is especially hard to follow if there is additional routing in views.

I'd prefer if mayberoute could go away in the future and each view decides for itself if it should be routed to `/` when the device is unplugged or not.
i.e. something like `import { routeToDefaultViewOnDeviceUnplug } from 'utils/routingHelper'` and  `this.unsubscribe = routeToDefaultViewOnDeviceUnplug();`

The device subscription probably makes sense that it stays in app.tsx but the accountlist fetching + subscription could be in a reusable component and only included in views that actually need it.

Reply:
> Personally I don't like mayberoute it was the cause of various routing bugs in the past and has to maintain a centralized list of exception routing for different views, it is especially hard to follow if there is additional routing in views.

Fyi, before we had maybeRoute, we had `route()` scattered and it also had many routing bugs and routing regressions. I had put in one place hoping that it would improve that.

But my comment here was actually not about maybeRoute() at all. Just an observation that for one `subscribe` decorator now, there are two very similar calls that have to be done now, e.g. `getDeviceList` and `syncDeviceList`. Not a problem for now.

Reply:
> Fyi, before we had maybeRoute, we had route() scattered and it also had many routing bugs and routing regressions. I had put in one place hoping that it would improve that.

👍 

For now, I think we should stick to the current way, devices routing done by app with a list of exceptions. 

Alternative would be to not route in app.tsx at all and make each view take care about the logic (this could be done with reusable modules to be DRY).